### ISSUE TITLE
"Uncaught (in promise)" error after failed declarative reflex

### ISSUE THREAD

**Issue by leastbad** [hopsoft/stimulus_reflex#170]
# Bug Report

When a declared (eg. `data-reflex`) reflex fails on the server, the promise rejection generates an error.

## To Reproduce

Vanilla Rails 6 app w/SR + scaffolded resource. Declare a reflex that goes to a non-existant action in a reflex class eg:

```html
<h1 data-reflex="click->ApplicationReflex#meat">MEAT</h1>
```

This generates the following log output:

```
StimulusReflex::Channel Failed to invoke ApplicationReflex#meat! http://localhost:3000/posts/new undefined method `meat' for class `ApplicationReflex' /home/leastbad/.rvm/gems/ruby-2.6.5/gems/stimulus_reflex-3.1.3/lib/stimulus_reflex/channel.rb:53:in `method'
```

and the following error in the console:

![uncaught](https://user-images.githubusercontent.com/38150464/80281354-7445a280-86d8-11ea-86b1-eec27156ecac.png)

## Expected behavior

No errors in the console.

### StimulusReflex

- Gem: 3.1.3
- Node package: 3.1.3

### External tools

- Ruby: 2.6.5
- Rails: 6.0.0.2
- Node: 13.10.0

### Browser

- Browser: Chrome
- Version: 81

**Comment by leastbad** [comment]
@hopsoft I'm trying to work through this https://discord.com/channels/629472241427415060/708177464894947362/710151681689518130

The question really is: assuming unnecessary scary errors are lazy APIs, should we / could we make an effort to determine whether to return a promise or not based on how `stimulate()` was called (via declared reflex or explicit JS)

**Comment by leastbad** [comment]
Closed by #193
